src/seedu/addressbook/data/AddressBook.java

```diff
@@ -83,6 +83,29 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
         syncTagsWithMasterList(toAdd);
     }
 
+    /**
+     * Edits a person in the address book.
+     *
+     * @throws DuplicatePersonException if an equivalent person already exists.
+     */
+    public void editPerson(ReadOnlyPerson target, Person replacement)
+            throws DuplicatePersonException, PersonNotFoundException {
+        addPerson(replacement);
```

Comment:
Is this the wanted behavior? **AddressBook** throws **DuplicatePersonException** if it finds a person with the same name and phone number in the list. Now what if I want to change John Doe's email address from _johnd@gmail.com_ to _johnd@u.nus.edu_? Then **AddressBook** will throw **DuplicatePersonException**. Should you add first then remove or should you remove first then add?

Reply:
Hmm, I didn't check that duplicate considers name and phone only... intention was to disallow editing a person to be exactly same as another. I'll look into using the other compare function instead.